bindings/kafka/metadata.yaml

```
@@ -0,0 +1,189 @@
+# yaml-language-server: $schema=../../component-metadata-schema.json
+schemaVersion: v1
+type: bindings
+name: kafka
+version: v1
+status: stable
+title: "Apache Kafka"
+urls:
+  - title: Reference
+    url: https://docs.dapr.io/reference/components-reference/supported-bindings/kafka/
+binding:
+  output: true
+  input: true
+  operations:
+    - name: create
+      description: "Publish a new message in the topic."
+authenticationProfiles:
+  - title: "OIDC Authentication"
+    description: |
+      Authenticate using OIDC.
+    metadata:
+      - name: authType
+        type: string
+        required: true
+        description: |
+          Configure or disable authentication.
+          The authType must be set to oidc.
+      - name: oidcTokenEndpoint
+        type: string
+        description: |
+          Full URL to an OAuth2 identity provider access token endpoint.
+          Required when authType is set to oidc.
+        example: '"https://identity.example.com/v1/token"'
+      - name: oidcClientID
+        description: |
+          The OAuth2 client ID that has been provisioned in the identity provider.
+          Required when authType is set to oidc.
+        example: '"dapr-kafka"'
+        type: string
+      - name: oidcClientSecret
+        type: string
+        sensitive: true
+        description: |
+          The OAuth2 client secret that has been provisioned in the identity provider.
+          Required when authType is set to oidc.
+        example: '"KeFg23!"'
+      - name: oidcScopes
+        type: string
+        description: |
+          Comma-delimited list of OAuth2/OIDC scopes to request with the access token.
+          Recommended when authType is set to oidc.
+        example: '"openid,kafka-prod"'
+        default: '"openid"'
+  - title: "SASL Authentication"
+    description: |
+      Authentication using SASL.
+    metadata:
+      - name: authType
+        type: string
+        required: true
+        description: |
+          Configure or disable authentication.
+          The authType must be set to "password".
+        example: '"none"'
+      - name: saslUsername
+        type: string
+        description: |
+          The SASL username used for authentication.
+          Only required if authType is set to "password".
+        example: '"myuser"'
+      - name: saslPassword
+        type: string
+        senstive: true
+        description: |
+          The SASL password used for authentication.
+          Only required if authType is set to "password".
+        example: '"mypassword"'
+      - name: saslMechanism
+        type: string
+        description: |
+          The SASL Authentication Mechanism you wish to use.
+          Only required if authType is set to "password".
```

From reading [the code](https://github.com/dapr/components-contrib/blob/master/internal/component/kafka/auth.go#L25)  it seems we do not actually require this field to be set. And while we don't really test for the `PLAINTEXT` string (it gets matched in the `else` branch), we do a exact string match for `"SHA-256"` and `"SHA-512"`. So better use `allowedValues` here too.

So
* required: false should be used
* Description should
  * mention that if unset defaults to `PLAINTEXT`.
  * not state it is required when it isn't.
*  `allowedValues` should be used


```suggestion
          Only required if authType is set to "password".
          If unset defaults to `PLAINTEXT`.
```